app/code/Magento/Weee/Test/Unit/Plugin/Catalog/Controller/Adminhtml/Product/Initialization/Helper/ProcessTaxAttributeTest.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Weee\Test\Unit\Plugin\Catalog\Controller\Adminhtml\Product\Initialization\Helper;
+
+use Magento\Catalog\Controller\Adminhtml\Product\Initialization\Helper;
+use Magento\Catalog\Model\Product;
+use Magento\Eav\Model\Entity\Attribute\AbstractAttribute;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Weee\Plugin\Catalog\Controller\Adminhtml\Product\Initialization\Helper\ProcessTaxAttribute;
+use PHPUnit\Framework\MockObject\Matcher\InvokedCount as InvokedCountMatcher;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class ProcessTaxAttributeTest extends TestCase
+{
+    /**
+     * Weee frontend input
+     */
+    private const WEEE_FRONTEND_INPUT = 'weee';
+
+    /**
+     * Text frontend input
+     */
+    private const TEXT_FRONTEND_INPUT = 'text';
+
+    /**
+     * Stub weee attribute code
+     */
+    private const STUB_WEEE_ATTRIBUTE_CODE = 'weee_1';
+
+    /**
+     * Stub weee attribute value
+     */
+    private const STUB_WEEE_ATTRIBUTE_VALUE = 1122;
+
+    /**
+     * @var ProcessTaxAttribute
+     */
+    private $plugin;
+
+    /**
+     * @var Helper|MockObject
+     */
+    private $subjectMock;
+
+    /**
+     * @var Product|MockObject
+     */
+    private $productMock;
+
+    /**
+     * @var Product|MockObject
+     */
+    private $resultMock;
+
+    /**
+     * Prepare environment for test
+     */
+    protected function setUp()
+    {
+        $this->subjectMock = $this->createMock(Helper::class);
+        $this->resultMock = $this->createMock(Product::class);
+        $this->productMock = $this->createMock(Product::class);
+
+        $objectManager = new ObjectManager($this);
+        $this->plugin = $objectManager->getObject(ProcessTaxAttribute::class);
+    }
+
+    /**
+     * Test afterInitializeFromData when attributes are empty
+     */
+    public function testAfterInitializeFromDataWhenAttributesAreEmpty()
+    {
+        $this->resultMock->expects($this->any())->method('getAttributes')
+            ->willReturn([]);
+
+        $this->resultMock->expects($this->never())->method('setData')->willReturnSelf();
```

Comment:
I guess the invocation result definition is redundant, in case the method is not expected to be called, isn't it?
```suggestion
        $this->resultMock->expects($this->never())->method('setData');
```